backend/src/oe/service/op-block-header.service.ts

```diff
@@ -0,0 +1,91 @@
+import Bluebird = require('bluebird');
+import bitcoinApi from '../../api/bitcoin/bitcoin-api-factory';
+import config from '../../config';
+import logger from '../../logger';
+import opBlockHeaderRepository from '../repositories/OpBlockHeaderRepository';
+
+export class OpBlockHeaderService {
+
+  async $saveBlockHeader(blockHeight: number): Promise<void> {
+    try {
+      // ignoring first 6 blocks
+      if (blockHeight <= 6) {
+        return;
+      }
+
+      // Only storing 6th block from current tip
+      const blockHash = await bitcoinApi.$getBlockHash(blockHeight - 6);
+      const {
+        height,
+        version,
+        previousblockhash,
+        merkle_root,
+        timestamp,
+        difficulty,
+        nonce,
+        chainwork,
+        mediantime,
+      } = await bitcoinApi.$getBlock(blockHash);
+
+      const { totalfee, subsidy } = await bitcoinApi.$getBlockStats(blockHash);
+
+      await opBlockHeaderRepository.$saveBlockHeaderInDatabase({
+        height,
+        version,
+        chainWork: chainwork,
+        medianTime: mediantime,
+        previousBlockHash: previousblockhash,
+        merkleRoot: merkle_root,
+        timestamp,
+        difficulty,
+        nonce,
+        reward: totalfee + subsidy,
+      });
+    } catch (error) {
+      logger.err(
+        `Error while saving block header ${blockHeight - 6}: ${error}`
+      );
+    }
+  }
+
+  public async $syncOlderBlockHeader(): Promise<void> {
+    try {
+      logger.debug('Syncing older block headers');
+
+      let currentSyncedBlockHeight =
+        await opBlockHeaderRepository.$getLatestBlockHeight();
+      const currentBlockHeight =
+        (await bitcoinApi.$getBlockHeightTip()) -
+        config.MEMPOOL.INITIAL_BLOCKS_AMOUNT;
+
+      logger.debug(
+        `currentSyncedBlockHeight: ${currentSyncedBlockHeight}, currentBlockHeaderHeight: ${currentBlockHeight}`
+      );
+
+      if (currentSyncedBlockHeight >= currentBlockHeight) {
+        logger.debug('Already synced block headers.');
+        return;
+      }
+
+      currentSyncedBlockHeight += 1;
+
+      logger.debug(
+        `Syncing block header from #${currentSyncedBlockHeight} to #${currentBlockHeight}`
+      );
+
+      while (currentBlockHeight >= currentSyncedBlockHeight) {
+        await Bluebird.map(
+          [...Array(10).keys()].map((i) => i + currentSyncedBlockHeight),
```

Comment:
```
[...Array( max(1, max(10, currentBlockHeight - currentSyncedBlockHeight ))).keys()].map((i) => i + currentSyncedBlockHeight)
```
?

Reply:
actually:
```
max(1, min(10, currentBlockHeight - currentSyncedBlockHeight ))
```
which defines concurrent jobs of count in  range [1;10]

Reply:
Done